src/operator/join.rs

```diff
@@ -0,0 +1,215 @@
+use crate::{
+    algebra::{finite_map::KeyProperties, IndexedZSet, MapBuilder, ZRingValue},
+    circuit::{
+        operator_traits::{BinaryOperator, Operator},
+        Circuit, Scope, Stream,
+    },
+};
+use std::{borrow::Cow, marker::PhantomData};
+
+impl<P, I1> Stream<Circuit<P>, I1>
+where
+    P: Clone + 'static,
+{
+    /// Apply [`Join`] operator to `self` and `other`.
+    pub fn join<K, V1, V2, V, W, F, I2, Z>(
```

Comment:
I feel dizzy only looking at this list. Should it be documented?

Reply:
It is, in the `Join` operator.

Reply:
I'd add a "See the `Join` operator for more info" note here just so it's a wee more discoverable

Reply:
Done